### PR TITLE
Add filtering of invalid request parameters.

### PR DESCRIPTION
**Pull request by EreMaijala**
Not quite sure if the delegator thing is right, I just mimicked it from Zend\Mvc\Console's config provider and it seems to work..

Examples of invalid search parameters:
- lookfor=test%00
- lookfor=test%C0%A7%C0

TODO
- [x] Run full test suite
- [x] Add note to changelog when merging

### CODE REVIEW

File: module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  OAI_Server
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php]: I think this package name should be changed. :-)

**Review by demiankatz — COMMENTED** [review]
I noticed one minor typo. I'll test this in more detail when I'm back in the office tomorrow (I'm taking the day off today, but I like to keep up with my email). Regarding the delegator factory, what you've done makes sense to me for the moment, though we'll have to remember to get rid of it later if we move away from Zend\Mvc\Console (since it is currently deprecated).

File: module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php]: Oops. I changed it to HTTP, but feel free to suggest better alternatives.

Comment (EreMaijala) [reply module/VuFind/src/VuFind/Http/PhpEnvironment/Request.php]: I think it'll be obvious that the delegator doesn't work when all the command line tools start to fail, but yes, needs to be taken care of.

**Comment by demiankatz** [comment]
@EreMaijala, the full test suite is passing on this, and the code looks reasonable to me. I'm not going to merge it until I get back next week, just in case I'm overlooking something... but I think it's ready to go!

**Comment by demiankatz** [comment]
I'm back, so this has been merged. Thanks again!

**Pull request merged**